coderd/database/migrations/migrate_test.go

```diff
@@ -129,3 +139,197 @@ func TestCheckLatestVersion(t *testing.T) {
+	}
+
+	// Folders in testdata/full_dumps represent fixtures for a full
+	// deployment of Coder.
+	matches, err := filepath.Glob(filepath.Join("testdata", "full_dumps", "*"))
+	require.NoError(t, err)
+	for _, match := range matches {
+		tests = append(tests, testCase{
+			name:     filepath.Base(match),
+			path:     match,
+			useStats: true,
+		})
+	}
+
+	// These tables are allowed to have zero rows for now,
+	// but we should eventually add fixtures for them.
+	ignoredTablesForStats := []string{
+		"audit_logs",
+		"git_auth_links",
+		"group_members",
+		"licenses",
+		"replicas",
+	}
+	s := &tableStats{s: make(map[string]int)}
+
+	// This will run after all subtests have run and fail the test if
+	// new tables have been added without covering them with fixtures.
+	t.Cleanup(func() {
+		emptyTables := s.Empty()
+		slices.Sort(emptyTables)
+		for _, table := range ignoredTablesForStats {
+			i := slices.Index(emptyTables, table)
+			if i >= 0 {
+				emptyTables = slices.Delete(emptyTables, i, i+1)
+			}
+		}
+		if len(emptyTables) > 0 {
+			t.Logf("The following tables have zero rows, consider adding fixtures for them or create a full database dump:")
+			t.Errorf("tables have zero rows: %v", emptyTables)
+			// TODO(mafredri): Placeholder URL.
+			t.Logf("See https://github.com/coder/coder/blob/main/migrations/README.md#creating-fixtures for more information")
+		}
+	})
+
+	for _, tt := range tests {
+		tt := tt
+
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			db := testSQLDB(t)
+
+			ctx, _ := testutil.Context(t)
+
+			// Prepare database for stepping up.
+			err := migrations.Down(db)
+			require.NoError(t, err)
+
+			// Initialize migrations for fixtures.
+			fDriver, fMigrate := setupMigrate(t, db, tt.name, tt.path)
+
+			nextStep, err := migrations.Stepper(db)
+			require.NoError(t, err)
+
+			var fixtureVer uint
+			nextFixtureVer, err := fDriver.First()
+			require.NoError(t, err)
+
+			for {
+				version, more, err := nextStep()
+				require.NoError(t, err)
+
+				if !more {
+					// We reached the end of the migrations.
+					break
+				}
+
+				if nextFixtureVer == version {
+					err = fMigrate.Steps(1)
+					require.NoError(t, err)
+					fixtureVer = version
+
+					nv, _ := fDriver.Next(nextFixtureVer)
+					if nv > 0 {
+						nextFixtureVer = nv
+					}
+				}
+
+				t.Logf("migrated to version %d, fixture version %d", version, fixtureVer)
+			}
+
+			// Gather number of rows for all existing tables
+			// at the end of the migrations and fixtures.
+			var tables pq.StringArray
+			err = db.QueryRowContext(ctx, `
+				SELECT array_agg(tablename)
+				FROM pg_catalog.pg_tables
+				WHERE
+					schemaname != 'information_schema'
+					AND schemaname != 'pg_catalog'
+					AND tablename NOT LIKE 'test_migrate_%'
+			`).Scan(&tables)
+			require.NoError(t, err)
+
+			for _, table := range tables {
+				var count int
+				err = db.QueryRowContext(ctx, "SELECT COUNT(*) FROM "+table).Scan(&count)
```

Comment:
A slightly uglier method than checking pg stats, problem with stats is we don't control when they're updated.